### PR TITLE
Groups by Numbers and Contributors

### PR DESCRIPTION
**Pull request by bdonald25**
#1852  Group by Numbers (For Pie Chart Only)
#2088 Group by Contributors

### CODE REVIEW

File: app/assets/javascripts/visualizations/highvis/bar.coffee [hunk 1]
```diff
@@ -142,8 +142,10 @@ $ ->
 
       drawControls: ->
         super()
-
-        @drawGroupControls(data.textFields)
+        # Remove group by number fields, only for pie chart
```
Comment (tylerpuleo) [inline app/assets/javascripts/visualizations/highvis/bar.coffee]: What's the rational behind adding the feature to base vis and then removing it from all other vises as opposed to just adding it to pie.coffee?

File: app/assets/javascripts/visualizations/highvis/highmodifiers.coffee [hunk 2]
```diff
@@ -278,15 +300,21 @@ $ ->
...
+          if data.fields[f].fieldID != -1
+            groups.push(data.fields[f].fieldName)
+      else
+        for dp in @dataPoints
+          if dp[gIndex] isnt null
+            result[String(dp[gIndex]).toLowerCase()] = true
 
-      groups = for keys of result
-        keys
+        groups = for keys of result
```
Comment (tylerpuleo) [inline app/assets/javascripts/visualizations/highvis/highmodifiers.coffee]: looks like an indentation error

File: app/assets/javascripts/visualizations/highvis/highmodifiers.coffee [hunk 3]
```diff
@@ -278,15 +300,21 @@ $ ->
...
+          keys
```
Comment (tylerpuleo) [inline app/assets/javascripts/visualizations/highvis/highmodifiers.coffee]: same here

**Comment by tylerpuleo** [comment]
Read my comments. Looks good otherwise. Migration was successful, none of the projects I used to test were broken on dev new or old.

**Comment by bdonald25** [comment]
Those indentations are intentional. Those lines are meant to be in the else part of the if statement. Those lines are getting the groups based on the data of the field that is being grouped by.  For example, the different names from the data in the data set name field. Group by numbers doesn't work like that, it gets its groups from the field names instead of the data of the field that is being grouped. 

Also, I am not adding it to base vis and then removing it from the other visualizations. The group by fields are treated like any other field. They are text fields. Before, all the visualizations would call @drawGroupControls(data.textFields). Instead of adding all the text fields to the group by menu, I am displaying all the text fields except group by numbers for visualizations other than pie.

**Comment by tylerpuleo** [comment]
Good to merge then

**Pull request merged**